model-optimizer/extensions/front/onnx/quantize_linear_resolver.py

```diff
@@ -0,0 +1,84 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import numpy as np
+
+from extensions.front.onnx.quantize_dequantize_linear import QuantizeDequantizeLinear
+from extensions.ops.Cast import Cast
+from extensions.ops.elementwise import Mul, Sub
+from extensions.ops.fakequantize import FakeQuantize
+from mo.front.common.partial_infer.utils import int64_array, float_array
+from mo.front.common.replacement import FrontReplacementOp
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.ops.broadcast import Broadcast
+from mo.ops.const import Const
+from mo.ops.shape import Shape
+from mo.utils.error import Error
+
+
+class QuantizeLinearResolver(FrontReplacementOp):
+    """
+    Replaces QuantizeLinear with FakeQuantize
+    """
+    op = "QuantizeLinear"
+    enabled = True
+
+    def run_after(self):
+        return [QuantizeDequantizeLinear]
+
+    def replace_op(self, graph: Graph, node: Node):
+        node_name = node.soft_get('name', node.id)
+
+        if node.is_in_port_connected(2):
+            zerop = node.in_port(2).get_source().node
+        else:
+            zerop = Const(graph, {'value': np.array(0, dtype=np.uint8), 'name': node_name + '/ZeroPoint'}).create_node()
+
+        # only constant for zero_point is supported
+        assert zerop.soft_get('type') == 'Const'
+        zero_point_type = zerop.value.dtype
+        # data type affects range of output values: [-128..127] or [0..255]
+        if zero_point_type == np.int8:
```

Comment:
If the zero_point input was not originally connected to the QuantizeLinear node we generate it ourselves in the line 48 and hardcode `dtype=np.uint8`. Here we check the zero point data type and specify different high/low values. Is this correct approach? May be we need in some case to specify the `dtype=np.int8` instead of `dtype=np.uint8` in line 48?

Reply:
Yes, this is a correct approach, because uint8 type is default as specified by ONNX spec.